addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1321,19 +1321,29 @@ const Wysiwyg = Widget.extend({
         }
 
         if (params.node) {
-            params.node.replaceWith(element);
+            const changedImage = params.node.tagName === element.tagName && element.tagName === 'IMG';
+            if (changedImage) {
+                // Instead of replacing an image with another one and
+                // recreating its snippet editor, its attributes are
+                // changed, and the snippet editor is kept.
+                for (const attribute of params.node.attributes) {
+                    params.node.setAttribute(attribute.nodeName, '');
+                }
+                for (const attribute of element.attributes) {
+                    params.node.setAttribute(attribute.nodeName, attribute.nodeValue);
+                }
```

Comment:
Just out of curiosity, does the media dialog actually sets up anything else than the 'src' attribute? :thinking:

Reply:
The media dialog will add attributes/classes specific to the selected media type, remove the ones specific to other media types (and keep the remaining ones).

Reply:
After checking a second time, my previous comment is wrong, the media dialog does not keep all the remaining attributes.
When replacing a media, it will:
- create a new element of the selected type, with the src attribute set,
- add back the classes and styles that were on the initial element, that are not specific to other media types,
- add back the shapes data atributes,
- for some media type, add extra attributes (the attachment description will be used as the alt attributes for images for example).

In the end, I think the current imp to optimize replacing images is not correct, the snippet editor and its associated options need to be recreated. Otherwise, there can be some weird behaviours like in 15.0:
- set the description of an image from the image options,
- replace the image,
=> the widget value in the image options still holds the old description value, when the image itself does not have any after being replaced.

Instead, I am checking if we can optimize by keeping the columns editor and options and recreate only the image's one, with something like
![image](https://user-images.githubusercontent.com/58561718/201954135-d2f61ce7-f4e6-4865-bddd-289b4520aeb5.png)